src/python/profiler/formatters/SimpleTableFormatter.py

```diff
@@ -0,0 +1,112 @@
+__author__ = 'jan-hybs'
+
+import re, os
+
+class SimpleTableFormatter (object) :
+
+
+    def __init__(self):
+        self.json = None
+        self.output = ""
+
+        self.headerCols = []
+        self.maxNameSize = 12
+
+        self.bodyRows = []
+        self.maxBodySize = None
+        self.headerFields = ("tag", "call count", "max time", "max/min time", "avg time", "total", "source")
+        self.styles = {"linesep": os.linesep, "padding": 0}
+        self.linesep = os.linesep
+
+    def set_styles (self, styles):
+        self.styles.update (styles)
+        self.linesep = self.styles["linesep"]
+
+    def format (self, json) :
+        self.json = json
+        self.processHeader (json)
+        self.processBody (json, 0)
+        self.maxBodySize = [n + self.styles['padding'] for n in self.maxBodySize]
+
+        lineDivider = (sum (self.maxBodySize) + 2 + len (self.maxBodySize) * 2) * "-"
+        fmtHead = "{:" + str (self.maxNameSize + 2) + "s}{}" + self.linesep
+
+        for pair in self.headerCols : self.output += fmtHead.format (*pair)
+
+        self.output += lineDivider
+        self.output += self.linesep
+        self.output += "| "
+        for i in range (len (self.headerFields)) :
+            self.output += ("{:^" + str (self.maxBodySize[i]+1) + "s}|").format (self.headerFields[i])
+        self.output += self.linesep
+        self.output += lineDivider
+        self.output += self.linesep
+
+        for tup in self.bodyRows :
+            self.output += "| "
+            fields = []
+            for i in range (len (self.maxBodySize)) :
+                fields.append (("{:" + tup[i][0] + "" + str (self.maxBodySize[i]) + "s}").format (tup[i][1]))
+            self.output += " |".join(fields)
+            self.output += " |" + self.linesep
+            # self.output += fmtBody.format (*tup)
+
+        self.output += lineDivider
+        return self.output
+
+
+    def appendToHeader (self, name, value=None) :
+        value = value if value is not None else self.json[name.lower ().replace (" ", "-")]
+        self.headerCols.append ((name, value))
+
+        if self.maxNameSize < len (str (name)) :
+            self.maxNameSize = len (str (name))
+
+    def appendToBody (self, values) :
+        self.bodyRows.append (values)
+
+        if self.maxBodySize == None :
+            self.maxBodySize = [11] * len (values)
+
+        # update max length
+        self.maxBodySize = [max (self.maxBodySize[i], len (str (values[i][1]))) for i in range (len (self.maxBodySize))]
```

Comment:
Break into meaningful subexprs. Shouldn't be values already converted to strings?